code/game/objects/random/subtypes/plants.dm

```diff
@@ -0,0 +1,24 @@
+/obj/random/seaweed
+	name = "random seaweed"
+	desc = "This is some random seaweed."
+	icon = 'icons/obj/structures/plants.dmi'
+	icon_state = "seaweed"
+
+/obj/random/seaweed/spawn_choices()
+	return list(
+		/obj/structure/flora/seaweed =       3,
+		/obj/structure/flora/seaweed/mid =   3,
+		/obj/structure/flora/seaweed/large = 2,
+		/obj/structure/flora/seaweed/glow =  1
+	)
+
+/obj/random/pottedplant
+	name = "random potted plant"
+	desc = "This is a random potted plant."
+	icon = 'icons/obj/structures/potted_plants.dmi'
+	icon_state = "plant-26"
+	spawn_nothing_percentage = 0
+	var/static/list/blacklisted_plants = list(/obj/structure/flora/pottedplant/unusual)
```

Comment:
Do note that this means you won't be able to modify this on subtypes. It might be better to make this a proc like `/obj/random/pottedplant/proc/blacklisted_plants()`, so that it can be overridden.

Reply:
Proc-level static vars can be overridden by overriding the proc.